maingate/src/range.rs

```diff
@@ -519,8 +532,9 @@ mod tests {
         let inputs = (2..20)
             .map(|number_of_limbs| {
                 let bit_len = LIMB_BIT_LEN * number_of_limbs + OVERFLOW_BIT_LEN;
+                let value = BigUint::from(1u32) << 16 - 1;
```

Comment:
```suggestion
                let value = BigUint::from(1u32) << bit_len - 1;
```
?

Alternatively, you can also do it like this I think: `F::from(2).pow(&[bit_len as u64, 0, 0, 0]) - F::one();`, then no conversion using `BigUint` necessary.